packages/pong-engine/src/state.ts

```diff
@@ -22,14 +22,16 @@ import {
   BALL_RADIUS,
 } from './constants';
 
+type EmptyPayload = object;
+
 type Act<Type extends string, Payload extends object> = {
   type: Type;
   payload: Payload;
 };
 
 export type Action =
-  | Act<'move', { deltaTime: number }>
-  | Act<'lose' | 'win' | GamePaddleMoveCommand, object>
+  | Act<'move' | 'addPoint', { deltaTime: number }>
```

Comment:
I have to say that this `Act` type is veeery confusing. Why are losePoint and GamePaddleMoveCommand (as ) together? Why is  'move' and 'addPoint' together?
I assume it's basically grouping all types that have the same payload, but it doesn't make it readable.

Reply:
Yes, they are together because they share the same type of payload. We can take a look to split them a bit